types/observer-mixin.d.ts

```diff
@@ -5,7 +5,11 @@
 
 import {Callback, PromiseOrVoid} from './common';
 
-export type Listener = (ctx: object, next: (err?: any) => void) => void;
+export interface OperationHookContext {
```

Comment:
Can we have API docs, example, r comments on what this new interface is for?  Same for the rest of changes.

Reply:
Added.